Issue 2466 - Palette colour SVG on list markers

Naaaaiix opened the pull request:
Fixes #2466 

Includes:
1. Automatic change of SVG icon content when changing the colour presets on SC on the editor.
2. Automatic update of the SC variable colours when rendering on frontend, which also updates SVG icon markers colour.
3. Added colour option for any type of list marker
4. Fixed reverse order error with custom ordered types
5. Fixed unordered list marker styles (was just showing disk)

Some notes:
1. Tests will be needed, @Alejandro9kgm set on your list onces this branch will be merge.
2. The current available icons for the list are the image shape ones; may we need a new category for list markers? @kyrapieterse @elzadj  what do you think?

Naaaaiix committed:
Add colour update on SVG icon markers list
Add colour palette backups updater on frontend
Fix dependency cycle on `getPaletteColor`

Naaaaiix committed:
Add colour to all list options
Fixes styles on unordenered lists

Naaaaiix committed:
Merge branch 'master' into issue-2466

Naaaaiix committed:
Fix order numbers on lists

Naaaaiix committed:
Enhanced Text Maxi list options

Naaaaiix committed:
Fix minor bug

Naaaaiix committed:
Add paragraph spacing for lists

kyrapieterse reviewed (approved):
APPPPPPPPPPPPPPPROVED!!!!!

elzadj reviewed (changes requested):
Great work! Just some minor bugs to fix.

The custom colour for svg resets when I change the icon:

https://user-images.githubusercontent.com/2401618/151344244-8da3f09d-df18-4d5c-b6cd-e549aca2b5e4.mp4

Naaaaiix committed:
Merge branch 'master' into issue-2466

Naaaaiix committed:
Fix SVG marker keeps color when replace
Fix custom organized lists when start was negative value
Fix responsive support
Fix list marker size unit bug
Fixes update color palette backups on server side afer merging master

elzadj reviewed (approved):
Everything is great, one note: the 'start from' acts kinda weird - if I start from 0, the first number must be 0, not 1. If I start from 1, the first one must be 1, not 2, etc.:

https://user-images.githubusercontent.com/2401618/151577878-14363e0a-2bc6-4cb7-896f-fafb31801db0.mp4

Naaaaiix committed:
Fixes list start issue

Naaaaiix committed:
Merge branch 'master' into issue-2466

Naaaaiix committed:
Update style.scss
